### ISSUE TITLE
[1.19.2 Forge Bug]: Active abilities stop working

### ISSUE THREAD

**Issue by transmascdragon** [DragonSurvivalTeam/DragonSurvival#430]
### ■ Your Discord ID or other contacts

@transmisogyny

### ■ Dragon Survival version and Modpack version (if have)

dragon survival 1.19.2 18.01.2024
[mods-1.zip](https://github.com/DragonSurvivalTeam/DragonSurvival/files/14291227/mods-1.zip)
list of mods ive also been using ^
forge 43.2.7 1.19.2

### ■ Single game or server?

Single (Default)

### ■ Do you delete old configs before writing a bug report?

I always delete configs

### ■ Bug Description and Reproduce

After playing in a world for awhile, abilities stop working. I press the ability activate button (default "c" key, there are no conflicting keybinds for "c") and the player animation for charging up the ability plays; but nothing else happens. When i create a brand new save (not recreating an already existing one), abilities do work perfectly fine, but at some point they stop working. I'm not entirely sure how or why this happens. The first time it happened it happened relatively early on but I can't say exactly when. the second time it happened a lot further in and I still cant say exactly what caused it to stop working unfortunately. I've also deleted both the server and client configs for the mod several times. Passive abilities seem to work fine.

https://github.com/DragonSurvivalTeam/DragonSurvival/assets/98481765/6245795a-959f-4e4c-bdfc-a1003864bc94
video of abilities not working^
https://www.dropbox.com/scl/fo/ehlagxas7tueersjrvy1d/h?rlkey=vuq02daogv5umtykgf37000zn&dl=0 second save this happened in, ended up deleting the first save

### ■ Expected behavior

I press the c key and whatever ability i have selected is activated. Which includes the charge up bar in the middle of the screen, depletion of whatever amount of mana each ability takes, whatever attack an ability might perform, and whatever effects an ability might give.

**Comment by DragonGirl42** [comment]
I have the same problem, and after doing extensive testing on my end to track down its cause, I'm still coming up with nothing. I created a new world with only Dragon Survival and GeckoLib, and I found that abilities indeed work. So I slowly added back the mountain of mods I use alongside it, a few at a time, and found that mods aren't the issue, as I could still use abilities after putting them all back.

I believe the issue has to do with Dragon Survival worlds created prior to one of the updates (probably version 18.10.2024). I tried deleting the Dragon Survival server config within the world files itself, but had no luck restoring abilities. I removed Dragon Survival from my mod selection entirely, opened the older world without it, then closed out and entered the world again with Dragon Survival added once more, but I was still unable to restore abilities. I even went and checked for updates to GeckoLib, but there are none for 1.19.2 past the one I'm already on.

Dragon Survival updated a few days ago, and the bug still hasn't been fixed. I'm worried that I'll have to try downgrading my version of Dragon Survival just to regain this essential gameplay mechanic, but if this doesn't get fixed anytime soon, I don't see another option.

**Comment by DragonGirl42** [comment]
I found something! After all my fiddling with world saves, version downgrades, and eventually gamerules, I made a breatkthrough - this bug seems linked with the gamerule **doDaylightCycle**. When it's set to false, active abilities don't work at all, even though their animations trigger as the dragon attempts to use them. However, when the gamerule is set to its default of true, abilities work as intended and expected.

I can only assume active abilities have some tie to the advancing of the day for their respective activation timers. When the day isn't advancing, neither do the timers.

THIS IS STILL A BUG! Dragon Survival _probably_ shouldn't be linking important mechanics to the time of day, especially with no indicator or warning that this is how abilities work, at least that I could find.
